src/script/lua_api/l_object.cpp

```diff
@@ -1586,44 +1586,212 @@ int ObjectRef::l_hud_get_hotbar_selected_image(lua_State *L)
+			throw LuaError("Skybox expects 6 textures.");
+		
+		if (!lua_isnil(L, -1))
+			sky_params.clouds          = getboolfield_default(L, -1,
+				"clouds", true);
+
+		if (sky_params.type != "custom") {
+			getServer(L)->setSky(player, sky_params);
+			lua_pushboolean(L, true);
+			return 1;
+		}
+	
+		if (!lua_isnil(L, -1))
+			sky_params.default_fog     = getboolfield_default(L, -1,
+				"default_fog", false);
+		if (!lua_isnil(L, -1))
+			sky_params.overlay_visible = getboolfield_default(L, -1,
+				"overlay_visible", false);
+
+		lua_getfield(L, 2, "sun"); // Moon related object management
+
+		sky_params.sun.visible      = getboolfield_default(L, -1, 
+			"visible", true);
+		sky_params.sun.yaw          = getfloatfield_default(L,-1,
+			"yaw", 90);
+		sky_params.sun.tilt         = getfloatfield_default(L, -1,
+			"tilt", 0);
+		sky_params.sun.texture      = getstringfield_default(L, -1,
+			"texture", "sun.png");
+		sky_params.sun.sunrise_glow = getboolfield_default(L, -1,
+			"sunrise_glow", true);
+		lua_pop(L, 1);
 
-	video::SColor bgcolor(255,255,255,255);
-	read_color(L, 2, &bgcolor);
+		lua_getfield(L, 2, "moon"); // Moon related object management
+
+		sky_params.moon.visible = getboolfield_default(L, -1, 
+			"visible", true);
+		sky_params.moon.yaw     = getfloatfield_default(L, -1, 
+			"yaw", -90);
+		sky_params.moon.tilt    = getfloatfield_default(L, -1, 
+			"tilt", 0);
+		sky_params.moon.texture = getstringfield_default(L, -1, 
+			"texture", "moon.png");
+		lua_pop(L, 1);
+
+		lua_getfield(L, 2, "stars"); //Stars related object management
+
+		sky_params.stars.visible = getboolfield_default(L, -1,
+			"visible", true);
+		sky_params.stars.yaw    = getfloatfield_default(L, -1,
+			"yaw", 0);
+		sky_params.stars.tilt   = getfloatfield_default(L, -1,
+			"tilt", 0);
+		sky_params.stars.number = getintfield_default(L, -1,
+			"count", 200);
+		lua_pop(L, 1);
+
+		lua_getfield(L, 2, "overlay_textures");
+
+		sky_params.overlay_textures.clear();
+		if (lua_isnil(L, -1)) {
+			sky_params.overlay_textures.emplace_back(
+				"__not_supplied__");
+		} else {
+			lua_pushnil(L);
+			while (lua_next(L, -2) != 0) {
+				// key at index -2 and value at index -1
+				if (lua_isstring(L, -1)) {
+					sky_params.overlay_textures.emplace_back(
+						readParam<std::string>(L, -1));
+				} else
+					sky_params.overlay_textures.emplace_back("");
+				// removes value, keeps key for next iteration
+				lua_pop(L, 1);
+			}
+		}
+		lua_pop(L, 1);
+
+		if (sky_params.overlay_textures.size() != 6 &&
+				sky_params.type == "custom" &&
+				sky_params.overlay_textures[0] !=
+				"__not_supplied__")
+			throw LuaError("Overlay textures expects 6 textures.");
+
+		getServer(L)->setSky(player, sky_params);
+		lua_pushboolean(L, true);
+		return 1;
+	} else { 
+		// handle deprecated set_sky here;
+		// we also use the brand new set_sky struct
+		log_deprecated(L,
+		"Deprecated call to set_sky, see lua_api.txt for details.");
```

Comment:
Missing indent. Would fit onto one line without violating the character limit.

Reply:
The indented line is 73 characters long.